src/cli/commands/force/lightning/lwc/preview.ts

```diff
@@ -96,98 +103,245 @@ export default class Preview extends Setup {
             });
     }
 
-    public validateAdditionalInputs(): Promise<void> {
-        const compName = this.flags.componentname;
-        return new Promise<void>((resolve, reject) => {
-            const isValid: boolean = compName
-                ? compName.trim().length > 0
-                : false;
-            if (isValid === false) {
-                this.logger.debug('Invoked validateComponentName in preview');
-                reject(
+    public async validateAdditionalInputs(): Promise<void> {
+        const platform = this.flags.platform;
+
+        const compName = CommandLineUtils.resolveFlag(
+            this.flags.componentname,
+            ''
+        );
+
+        const targetApp = CommandLineUtils.resolveFlag(
+            this.flags.targetapp,
+            PreviewUtils.BROWSER_TARGET_APP
+        );
+
+        const configFile = CommandLineUtils.resolveFlag(
+            this.flags.configfile,
+            ''
+        );
+
+        const hasConfigFile = configFile.trim().length > 0;
+
+        const isBrowserTargetApp = PreviewUtils.isTargetingBrowser(targetApp);
+
+        const isValidCompName = compName.trim().length > 0;
+
+        this.logger.debug('Validating Preview command inputs.');
+
+        // check if user provided a config file when targetapp=browser
+        // and warn them that the config file will be ignored.
+        if (isBrowserTargetApp && hasConfigFile) {
+            this.logger.warn(
+                messages.getMessage('ignoringConfigFileFlagDescription')
+            );
+        }
+
+        if (
+            isValidCompName === false ||
+            (isBrowserTargetApp === false && hasConfigFile === false)
+        ) {
+            return Promise.reject(
+                new SfdxError(
+                    messages.getMessage('error:invalidInputFlagsDescription'),
```

Comment:
Given our increasing proliferation of different flag configurations, I'd like us to split out specific violation messages, so consumers don't need to hunt for the root cause. Let's make a separate message for invalid (well, missing really) component names and config files.